src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -597,13 +638,16 @@ impl OpenIDConnectAuthProvider {
 }
 
 impl AuthProvider for OpenIDConnectAuthProvider {
-    // TODO: handle error responses from the provider as per RFC 6749 and OpenID
     // Connect Core 1.0 section 3.1.26 Authentication Error Response
+    // OAuth 2.0 RFC-674 4.1.2.1 (Authorization Request Errors) & 5.2 (Access Token Request Errors)
 
+    /// Tries to extend the session with the OIDC Provider once or multiple times (retries and/or back-off).
```

Comment:
This comment isn't quite correct. This fn is also responsible for completing the handshake for initial creation of a session, not just for attempting to extend an existing session.